### PR TITLE
refactoring to use c# 7 features

### PR DESCRIPTION
**Pull request by raskolnikoov**
refactoring to use c# features (tuples,  type matching). saturday fun.. :)

### CODE REVIEW

File: src/Proto.Persistence/Persistence.cs [hunk 1]
```diff
@@ -23,17 +23,15 @@ public class Persistence
...
             {
-                Index = t.Item2;
-                await Context.ReceiveAsync(new RecoverSnapshot(t.Item1));
-            }
+                Index = snapshot.Index;
+                await Context.ReceiveAsync(new RecoverSnapshot(snapshot.Data));
+            };
 
-            await State.GetEventsAsync(Name, Index, callbackData =>
+            await State.GetEventsAsync(Name, Index, Callback =>
```
Comment (tomliversidge) [inline src/Proto.Persistence/Persistence.cs]: i preferred callbackData as it is the data here, not the callback

File: src/Proto.Persistence/Persistence.cs [hunk 2]
```diff
@@ -67,11 +65,10 @@ public class Persistence
                 switch (context.Message)
                 {
                     case Started _:
-                        var p = context.Actor as IPersistentActor;
-                        if (p != null)
+                        if(context.Actor is IPersistentActor actor)
```
Comment (tomliversidge) [inline src/Proto.Persistence/Persistence.cs]: nice!

**Comment by tomliversidge** [comment]
Note this will clash with https://github.com/AsynkronIT/protoactor-dotnet/pull/115 as that pull request does not use the new named-tuple providerState.GetSnapshotAsync() api. If this is merged first, I can rebase the other PR to include these changes. Will also affect @jrouaix work on EventStore provider

File: src/Proto.Persistence/Persistence.cs (hunk 1 above)
Comment (raskolnikoov) [inline src/Proto.Persistence/Persistence.cs]: reverted :)

**Comment by raskolnikoov** [comment]
Merging..

**Pull request merged**